projects/ui/src/hooks/farmer/useFertilizerSummary.ts

```diff
@@ -14,35 +13,146 @@ import { USDC } from '~/constants/tokens';
  * @param tokens Token form state.
  */
 export default function useFertilizerSummary(
-  tokens: FormTokenState[]
+  tokens: FormTokenStateNew[]
 ) {
-  const Usdc = useChainConstant(USDC);
+  const sdk = useSdk();
+  const usdc = sdk.tokens.USDC;
+  const eth = sdk.tokens.ETH;
   const [humidity] = useHumidity();
-  const summary = tokens.reduce((agg, curr) => {
-    const amount = (
-      curr.token === Usdc
-        ? curr.amount
-        : curr.amountOut
-    );
-    if (amount) {
-      agg.usdc = agg.usdc.plus(amount);
-      if (curr.amount && curr.amountOut) {
-        agg.actions.push({
-          type: ActionType.SWAP,
-          tokenIn: curr.token,
-          tokenOut: Usdc,
-          amountIn: curr.amount,
-          amountOut: curr.amountOut,
-        });
+
+  // const summary = useMemo(() => {
+  //   const tokenActionMap: { [token: string]: SwapAction } = {};
+  //   return tokens.reduce((memo, curr, i) => {
+  //     const amount = (
+  //       usdc.equals(curr.token)
+  //         ? curr.amount
+  //         : curr.amountOut
+  //     );
+
+  //     if (amount) {
+  //       memo.usdc = memo.usdc.plus(amount);
+  //       const tokenKey = curr.token.equals(eth) ? 'eth' : curr.token.address;
+
+  //       if (curr.amount && curr.amountOut) {
+  //         if (tokenKey in tokenActionMap) {
+  //           const existing = tokenActionMap[tokenKey];
+  //           tokenActionMap[tokenKey] = {
+  //             ...existing,
+  //             amountIn: existing.amountIn.plus(amount),
+  //             amountOut: existing.amountOut.plus(curr.amountOut),
+  //           };
+  //         } else {
+  //           tokenActionMap[tokenKey] = {
+  //             type: ActionType.SWAP,
+  //             tokenIn: getNewToOldToken(curr.token),
+  //             tokenOut: getNewToOldToken(usdc),
+  //             amountIn: curr.amount,
+  //             amountOut: curr.amountOut,
+  //           }
+  //         }
+  //       }
+  //     }
+
+  //     if (i === tokens.length - 1) {
+  //       memo.actions = Object.values(tokenActionMap);
+  //     }
+      
+  //     return memo;
+  //   }, {
+  //     usdc:     new BigNumber(0),  // The amount of USDC to be swapped for FERT.
+  //     fert:     new BigNumber(0),
+  //     humidity: humidity,          //
+  //     actions:  [] as Action[],    // 
+  //   })
+  // }, [humidity, tokens, eth, usdc]);
```

Comment:
Is there a reason for preserving this? If not, could we delete it? Same question for other commented out code in this file